src/variant.h

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2017-2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_VARIANT_H
+#define BITCOIN_VARIANT_H
+
+#include <utility>
+#include <type_traits>
+
+#include <boost/variant.hpp>
+
+//! Substitute for C++17 std::variant
+template <class... Types>
+class Variant
+{
+private:
+    boost::variant<Types...> m_var;
+
+public:
+    Variant()
+    {
+        m_var = {};
+    }
+    Variant(const Variant& other)
+    {
+        m_var = other.m_var;
+    }
+    Variant(Variant&& other)
+    {
+        m_var = std::move(other.m_var);
+    }
+    // Require that typeof(T) != Variant. to prevent wrong template instantiations
+    template <class T, typename = typename std::enable_if<!std::is_same<typename std::remove_reference<T>::type, Variant>::value>::type>
```

Comment:
In commit "Add a new Variant type" (2eacf26ff61a63fce088600ced16ec2d4d5d771d)

Might be a little better to replace `std::remove_reference` with `std::decay` to remove CV qualifiers as well

Reply:
ah ha! I looked for the closest thing to C++20's `std::remove_cvref` looks like `decay` is the better option